File: app/src/main/java/com/example/muzik/response_model/Artist.java
package com.example.muzik.response_model;

import java.util.ArrayList;
import java.util.List;

public class Artist {
    private long artistID;
    private String name;
    private List<Song> songs;
    private String imageURL;

    public Artist() {
    }

    public Artist(long artistID, String name) {
        this.artistID = artistID;
        this.name = name;
        songs = new ArrayList<>();
    }

    public Artist(long artistID, String name, List<Song> songs, String imageURL) {
        this.artistID = artistID;
        this.name = name;
        this.songs = songs;
        this.imageURL = imageURL;
    }

    public long getArtistID() {
        return artistID;
    }

    public void setArtistID(long artistID) {
        this.artistID = artistID;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Song> getSongs() {
        return songs;
    }

    public void setSongs(List<Song> songs) {
        this.songs = songs;
    }

    public String getImageURL() {
        return imageURL;
    }

    public void setImageURL(String imageURL) {
        this.imageURL = imageURL;
    }
}

File: app/src/main/java/com/example/muzik/data_model/standard_model/Artist.kt
package com.example.muzik.data_model.standard_model

import android.net.Uri
import com.example.muzik.data_model.api_model.response.ArtistResponse
import com.example.muzik.music_service.LocalMusicRepository

class Artist(
    val name: String? = "",
    val artistID: Long = -1,
    val imageURI: Uri? = null
) : Model {
    val listSongId: MutableList<Long> = ArrayList()

    fun requireName(): String {
        checkNotNull(name) {
            ("Artist $this must have non-null name")
        }
        return name
    }

    companion object {
        fun buildOnline(artistResponse: ArtistResponse): Artist {
            return Artist(
                name = artistResponse.name,
                artistID = artistResponse.artistID,
                imageURI = Uri.parse(artistResponse.imageURL)
            )
        }

        fun buildLocal(artistID: Long, name: String): Artist {
            return Artist(artistID = artistID, name = name)
        }
    }

    fun getSongs(): List<Song> {
        val listSong: MutableList<Song> = ArrayList()
        for (songId: Long in listSongId) {
            listSong.add(LocalMusicRepository.getSong(songId) as Song)
        }
        return listSong
    }
}